service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -581,11 +601,64 @@ public WayForPayOrderResponse saveFullOrderToDB(OrderResponseDto dto, String uui
 
         if (sumToPayInCoins <= 0 || !dto.isShouldBePaid()) {
             return getPaymentRequestDto(order, "");
-        } else {
-            PaymentRequestDto requestDto = formPaymentRequest(order.getId(), sumToPayInCoins);
-            String link = getLinkFromWayForPayCheckoutResponse(wayForPayClient.getCheckOutResponse(requestDto));
-            return getPaymentRequestDto(order, link);
         }
+
+        return processPayment(dto, order, sumToPayInCoins, currentUser);
+    }
+
+    private PaymentSystemResponse processPayment(OrderResponseDto dto, Order order, long sumToPayInCoins,
+        User currentUser) {
+        return switch (dto.getPaymentSystem()) {
+            case WAY_FOR_PAY -> processWayForPay(order, sumToPayInCoins);
+            case MONOBANK -> processMonoBank(order, sumToPayInCoins, currentUser);
+        };
+    }
+
+    private PaymentSystemResponse processWayForPay(Order order, long sumToPayInCoins) {
+        PaymentWayForPayRequestDto requestDto = formPaymentRequestForWayForPay(order.getId(), sumToPayInCoins);
+        String link = getLinkFromWayForPayCheckoutResponse(wayForPayClient.getCheckOutResponse(requestDto));
+        return getPaymentRequestDto(order, link);
+    }
+
+    private PaymentSystemResponse processMonoBank(Order order, long sumToPayInCoins, User currentUser) {
+        MonoBankPaymentRequestDto requestDto =
+            formPaymentRequestForMonoBank(order.getId(), sumToPayInCoins, currentUser);
+        CheckoutResponseFromMonoBank checkoutResponse = monoBankClient.getCheckoutResponse(requestDto, token);
+
+        return getPaymentRequestDto(order, checkoutResponse.getPageUrl());
+    }
+
+    private MonoBankPaymentRequestDto formPaymentRequestForMonoBank(Long orderId, long sumToPayInCoins, User user) {
+        Order order = orderRepository.findById(orderId)
+            .orElseThrow(() -> new NotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST));
+
+        return MonoBankPaymentRequestDto.builder()
+            .amount(1)
+            // .amount((int) sumToPayInCoins)
```

Comment:
do we need this comment?

Reply:
This is a temporary comment, for payment testing